### PR TITLE
sail 1.0.1 (new formula)

### PR DESCRIPTION
**Pull request by nhooyr**
- [x]  Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/sail.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    bin.install "bin/sail" => "sail"
+  end
+
+  def caveats; <<~EOS
+    You will need docker to use sail.
+    Please install via:
+
+      brew cask install docker
+  EOS
+  end
```
Comment (zbeekman) [inline Formula/sail.rb]: This caveat is not unique to homebrew. You will need docker on all OSes for Sail, no?

Unless I am mistaken, then this caveat should be removed.

File: Formula/sail.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  test do
+    output = `#{bin}/sail --help 2>&1`
+    output.include? "A utility for managing Docker-based code-server environments."
```
Comment (zbeekman) [inline Formula/sail.rb]: Is there anyway to test this functionality better than parsing the output of help?

Couldn't you add a `depends_on "docker" => :test` and then do something in the test with docker?

**Review by zbeekman — CHANGES_REQUESTED** [review]

**Comment by zbeekman** [comment]
I'd love to see this added. Let's see if we can't get creative for some slightly better tests.

File: Formula/sail.rb (hunk 2 above)
Comment (nhooyr) [reply Formula/sail.rb]: Will try.

**Comment by SMillerDev** [comment]
How about testing for an error that happens without docker?

**Comment by nhooyr** [comment]
May pick this up later.

**Pull request closed without merging**